app/views/components/LinkWithAccessibilityText.scala.html

```diff
@@ -0,0 +1,21 @@
+@*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@this()
+
+@(text: String, hiddenText: String, url: String, classes: String = "govuk-link govuk-!-font-weight-bold")
+
+<a class="@classes" href="@url">@text<span class="govuk-visually-hidden"> @hiddenText</span></a>
```

Comment:
so where possible we try not to create extra components - it's not necessarily the right decision - but it was something we decided at the start of coding on this project to stop us from having loads of endless components if they were really tiny, so just put this directly in the code

Reply:
I wasn't too sure on this, as I was trying to keep away with using raw html in strings in code when building the table rows - and this meant I could keep it consistent with using the govUkTag to build the row as seen here:

https://github.com/hmrc/trader-goods-profiles-frontend/pull/513/files#diff-b26544f058240ec1354778d425e5453dc4f3293811f42a9ee4745978108b49d3R34

Otherwise the below line

https://github.com/hmrc/trader-goods-profiles-frontend/pull/513/files#diff-6eb1f2f2b5fc231190fcf32039957fd5537a4ca543c670cb65e0e033c135de49R62

Will become

```
val fileLink = 
s"""<a class="govuk-link govuk-!-font-weight-bold" href="${data.downloadUrl}">
$messages("fileManagement.availableFiles.downloadText")
<span class="govuk-visually-hidden">
$messages("fileManagement.availableFiles.downloadText.hidden", fileCreated)
</span>
</a>"""
 ```
If that's fine/the team's approach then I am happy with refactoring to that